core/tests/protractor_desktop/blogDashboard.js

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+// //
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// //
+//      http://www.apache.org/licenses/LICENSE-2.0
+// //
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for the blog Dashboard page.
+ */
+
+var action = require('../protractor_utils/action.js');
+var forms = require('../protractor_utils/forms.js');
+var users = require('../protractor_utils/users.js');
+var waitFor = require('../protractor_utils/waitFor.js');
+var general = require('../protractor_utils/general.js');
+
+var BlogDashboardPage = require('../protractor_utils/BlogDashboardPage.js');
+
+describe('Blog dashboard functionality', function() {
+  var blogDashboardPage = null;
+
+  beforeAll(async function() {
+    blogDashboardPage = (
+      new BlogDashboardPage.BlogDashboardPage());
+    await users.createUserWithRole(
+      'blog@blogDashboard.com',
+      'blog',
+      'blog admin');
+    await users.login('blog@blogDashboard.com');
+  });
+
+  beforeEach(async function() {
+    await blogDashboardPage.get();
+  });
+
+  it('should check that editor name and profile photo is visible',
+    async function() {
+      await blogDashboardPage.expectCurrUserToHaveProfilePhoto();
+      await blogDashboardPage.expectCurrUsernameToBeVisible();
+    });
+
+  it('should create and delete a newly created blog post', async function() {
+    await blogDashboardPage.createNewBlogPost();
+    await blogDashboardPage.navigateToBlogDashboardPage();
+    await blogDashboardPage.expectNumberOfDraftsBlogPostsToBe(1);
+
+    await blogDashboardPage.deleteBlogPostWithIndex(0);
+    await blogDashboardPage.expectNumberOfDraftsBlogPostsToBe(0);
+  });
+
+  it('should check that blog post editor loads user profile',
+    async function() {
+      await blogDashboardPage.createNewBlogPost();
+
+      await blogDashboardPage.expectCurrUserToHaveProfilePhoto();
+      await blogDashboardPage.expectCurrUsernameToBeVisible();
+
+      await blogDashboardPage.navigateToBlogDashboardPage();
+      await blogDashboardPage.deleteBlogPostWithIndex(0);
+      await blogDashboardPage.expectNumberOfDraftsBlogPostsToBe(0);
+    });
+
+  it('should create, edit and delete a blog post from blog dashboard',
```

Comment:
This test seems to largely overlap with the previous one. If you just added some expectations, this one could replace it.

Since the e2e tests are relatively expensive to run, we want to keep them as efficient as possible by avoiding duplication between tests. Ditto for other tests

Reply:
Done.

Reply:
I still think there's some room to consolidate tests. For example, this test and the test before it could be combined.

ditto for the other tests

Reply:
Done.